src/Package.zig

```diff
@@ -124,3 +124,13 @@ pub fn addAndAdopt(parent: *Package, gpa: Allocator, name: []const u8, child: *P
     child.parent = parent;
     return parent.add(gpa, name, child);
 }
+
+pub fn getName(pkg: *const Package) []const u8 {
```

Comment:
I think it'd be better for `Package` to just store it's own name.

Reply:
This solution is already used in Sema to output errors, so I assumed this was an intentional decision to reduce the size of `Package` (small DOD optimization).

Reply:
> This solution is already used in Sema to output errors

And it is fragile there and leads to incorrect error messages. I don't think it makes sense to optimize the size of `Package` too much since aren't that many of them compared to other types. And since package names shouldn't allow null bytes you can use `[*:0]const u8` for the name, path and the keys in the hash map and end up with less memory usage.

Reply:
Oh, interesting, I didn't know it had caused issues. I'll take a look at refactoring this.

Reply:
I'm reading through the similar code in `Sema` (11665 to 11679) to look at switching it over to a stored name in `Package` and I'm a little bit confused. There, we have three cases: the package name is `"root"` (since it's `main_pkg` or `root_pkg` - what's the difference?), the package has a parent and so a name within that parent, or the fallback where the package has no name. The third case is what's confusing me: when does it happen? Should it ever be possible, or was it just added as a "just in case" fallback?

Reply:
`main_pkg` is what is given on the command line, `root_pkg` is the same as `main_pkg` for `build-{exe,obj,lib}` but `test_runner.zig` for `test`. The third case should be unreachable.

Reply:
Would it make sense to give the main package a different name in output to differentiate it from the test runner?

Reply:
I'd leave it as is until #12201 is resolved.